tests/unit/drtsans/test_i_of_q_annular_binning.py

```diff
@@ -108,11 +109,91 @@ def test_1d_flat_data():
     # verify the results
     assert binned_iq.intensity.size == 18
     np.testing.assert_equal(binned_iq.intensity, 1.0)
-    # uncertainties are proportional to the number of input bins contributing to the annullar wedge
+    # uncertainties are proportional to the number of input bins contributing to the annular wedge
     np.testing.assert_allclose(binned_iq.error, 0.6, atol=0.11)
     # actually the azimuthal angle
     np.testing.assert_equal(binned_iq.mod_q, np.arange(10.0, 360.0, 20.0))
 
 
+def test_1d_flat_data_wl():
+    """Test annular binning with and without wavelength bins"""
+    # Q2D data for one wavelength
+    data2d = IQazimuthal(
+        intensity=np.full((21, 21), 1.0, dtype=float),
+        error=np.full((21, 21), 1.0, dtype=float),
+        qx=np.arange(-10, 11, 1, dtype=float),
+        qy=np.arange(-10, 11, 1, dtype=float),
+    )
+    data2d = data2d.ravel()
+
+    # repeat Q2D data 3 times for 3 wavelengths
+    wl_uniq = [1.5, 2.5, 3.5]
+    num_wl = len(wl_uniq)
+    data2d_wl = IQazimuthal(
+        intensity=np.tile(data2d.intensity, num_wl),
+        error=np.tile(data2d.error, num_wl),
+        qx=np.tile(data2d.qx, num_wl),
+        qy=np.tile(data2d.qy, num_wl),
+        wavelength=np.repeat(wl_uniq, len(data2d.intensity)),  # [1.5, 1.5, ..., 2.5, 2.5, ..., 3.5, 3.5, ...]
+    )
+
+    # perform the annular binning
+    phi_binning = BinningParams(0, 360, 18)  # 20 deg bins
+    binned_i1d_wl = bin_annular_into_q1d(
+        data2d_wl, phi_binning, q_min=9.0, q_max=10.0, method=BinningMethod.NOWEIGHT, wavelength_bins=None
+    )
+    binned_i1d_no_wl = bin_annular_into_q1d(
+        data2d_wl, phi_binning, q_min=9.0, q_max=10.0, method=BinningMethod.NOWEIGHT
+    )
+
+    # verify the results
+    assert binned_i1d_wl.intensity.shape[0] == phi_binning.bins * num_wl
+    assert binned_i1d_no_wl.intensity.shape[0] == phi_binning.bins
+    # intensity
+    np.testing.assert_equal(binned_i1d_wl.intensity, 1.0)
+    np.testing.assert_equal(binned_i1d_no_wl.intensity, 1.0)
+    # uncertainties are inversely proportional to the number of input bins contributing to the annular wedge
+    np.testing.assert_allclose(binned_i1d_wl.error, 0.6, atol=0.11)
+    np.testing.assert_allclose(binned_i1d_no_wl.error, 0.4, atol=0.12)
+    # actually the azimuthal angle
+    np.testing.assert_equal(binned_i1d_wl.mod_q, np.tile(np.arange(10.0, 360.0, 20.0), num_wl))
+
+
+def test_1d_wavelength():
+    # Define input Q2D data
```

Comment:
can you please add a docstring describing what the test does? For instance, how does it differ from test `test_1d_flat_data_wl`?